core/app/models/spree/order.rb

```diff
@@ -844,7 +844,7 @@ def require_email
     end
 
     def ensure_inventory_units
-      if has_checkout_step?("delivery")
+      if Spree::Config.build_shipment_predicate_class.call(self)
```

Comment:
#### This comment is not blocking.

Maybe it's just me but I find this preference naming a bit complex to understand. For me it's not immediately clear what it does and what's the purpose of this preference. Maybe it would be better if it was wrapped into a `Spree::Order#needs_shipments_building?` or something similar? What do you think?

Reply:
I agree with this. And as changing configuration options in later releases is hard, we should agree on a better name before merging this.